cmd/gb-vendor/vendor/repo.go

```diff
@@ -137,13 +134,22 @@ func (g *gitrepo) URL() string {
 	return g.url
 }
 
-func (g *gitrepo) Clone() (WorkingCopy, error) {
+func (g *gitrepo) Checkout(branch, revision string) (WorkingCopy, error) {
 	dir, err := mktmp()
 	if err != nil {
 		return nil, err
 	}
 
-	if err := runOut(os.Stderr, "git", "clone", g.url, dir); err != nil {
+	args := []string{
+		"clone",
+		g.url,
+		dir,
+	}
+	if branch != "" {
+		args = append(args, "--branch", branch)
```

Comment:
I would also add `--single-branch`, that will clone only that specified branch. We don't need other branches and that will reduce size of the downloaded(cloned) repo.